trin-execution/src/trie_walker/filter.rs

```diff
@@ -1,120 +1,131 @@
+    pub fn new(slice_index: u16, slice_count: u16) -> Self {
+        assert!(
+            slice_index < slice_count,
+            "slice_index must be less than slice_count"
+        );
+
         // if slice_count is 0 or 1, we want to include the whole trie
         if slice_count == 0 || slice_count == 1 {
             return Self {
-                start_prefix: U256::ZERO,
-                end_prefix: U256::MAX,
+                start: U256::ZERO,
+                end: U256::MAX,
             };
         }
 
-        let slice_size = U256::MAX / U256::from(slice_count);
-        let random_slice_index = thread_rng().gen_range(0..slice_count);
+        let slice_size = U256::MAX / U256::from(slice_count) + U256::from(1);
 
-        let start_prefix = U256::from(random_slice_index) * slice_size;
-        let end_prefix = if random_slice_index == slice_count - 1 {
+        let start = U256::from(slice_index) * slice_size;
+        let end = if slice_index == slice_count - 1 {
             U256::MAX
         } else {
-            start_prefix + slice_size - U256::from(1)
+            start + slice_size - U256::from(1)
         };
 
-        Self {
-            start_prefix,
-            end_prefix,
-        }
+        Self { start, end }
     }
 
-    fn partial_prefix(prefix: U256, shift_amount: usize) -> B256 {
-        prefix
-            .arithmetic_shr(shift_amount)
-            .wrapping_shl(shift_amount)
-            .into()
+    pub fn random(slice_count: u16) -> Self {
+        // if slice_count is 0 or 1, we want to include the whole trie
+        if slice_count == 0 || slice_count == 1 {
+            return Self {
+                start: U256::ZERO,
+                end: U256::MAX,
+            };
+        }
+
+        Self::new(thread_rng().gen_range(0..slice_count), slice_count)
     }
 
-    pub fn is_included(&self, path: &[u8]) -> bool {
+    /// Check if a path is included in the filter
+    pub fn contains(&self, path: &[u8]) -> bool {
         // we need to use partial prefixes to not artificially exclude paths that are not exactly
         // the same length as the filter
         let shift_amount = 256 - path.len() * 4;
-        let partial_start_prefix = Filter::partial_prefix(self.start_prefix, shift_amount);
-        let partial_end_prefix = Filter::partial_prefix(self.end_prefix, shift_amount);
-        let path = partial_nibble_path_to_right_padded_b256(path);
+        let partial_start_prefix = (self.start >> shift_amount) << shift_amount;
+        let partial_end_prefix = (self.end >> shift_amount) << shift_amount;
+        let path = nibbles_to_right_padded_b256(path);
 
-        (partial_start_prefix..=partial_end_prefix).contains(&path)
+        (partial_start_prefix..=partial_end_prefix).contains(&path.into())
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use alloy::hex::FromHex;
-
     use super::*;
 
     #[test]
     fn test_new_random_filter() {
```

Comment:
nit: names of tests usually don't start with `test_` (it's redundant since they are already inside `tests` module)